Monika-After-Story/MonikaModDev
# New name set and removal of mom names

closes #5607 

This PR addresses an inconsistency between the player name topic and Monika nickname topic. It was possible to call Monika names which she wouldn't really be comfortable calling you, and as such, makes sense that she wouldn't be comfortable being called them herself.

Additionally, if you have any good/bad/awkward names to add, please either push directly to the branch if you have permissions, or if not, just add review comments to add them.

## Changes:
- Addition of an `awkward_nickname_list` for Monika nickname, which doesn't boot you to apologize and out of the nickname loop (another try again flow)
- Change the names of the player nickname global lists and regex compiled lists
- Added grandfathered nicknames for those which *were* fine before and are now considered awkward. This allows the specific nickname to be kept and still be considered good.
- Removed the nickname lists created by the `monika_affection_nickname` topic. We now just use the single lists created in `script-story-events.rpy` for the player name topics. (Split gender related things (monika nickname vs player nickname) using modifier lists
- Changed `persistent._mas_called_moni_a_bad_name` into a player model variable.
- Alpha ordered the nickname lists
- Fixed some ambiguous regex which would have lead to false positives in some of the searches
- Removed confirmation on the player name change topic as it was redundant.

## Testing:
- Verify both Monika name change and player name change topics still work as they did, detecting good/bad/awkward names.
- Verify that setting `persistent._mas_monika_nickname` to an item in the awkward nickname list and calling `v0_11_1` sets `persistent._mas_grandfathered_nickname` to that name and it is still allowed to be used as Monika's nickname via the affection nickame topic.
- Verify no fpyos
